options.go

```diff
@@ -93,3 +98,136 @@ func WithPreviousImage(name string) func(*ImageOptions) {
+
+type PushOption func(*IndexPushOptions) error
+
+type IndexPushOptions struct {
+	IndexFormatOptions
+	IndexRemoteOptions
+	Purge bool
+	Tags  []string // Tags with which the index should be pushed to registry
+}
+
+type IndexFormatOptions struct {
+	Format types.MediaType // The media type for the index (oci or docker)
+}
+
+type IndexRemoteOptions struct {
+	Insecure bool
+}
+
+type IndexOptions struct {
+	XdgPath                string
+	BaseImageIndexRepoName string
+	KeyChain               authn.Keychain
+	IndexFormatOptions
+	IndexRemoteOptions
+
+	// These options must be specified in each implementation's image index constructor
+	BaseIndex v1.ImageIndex
+}
+
+// IndexOptions
+
+// FromBaseImageIndex loads the ImageIndex at the provided path for the working image index.
+// If the index is not found, it does nothing.
+func FromBaseImageIndex(name string) func(*IndexOptions) error {
+	return func(o *IndexOptions) error {
+		o.BaseImageIndexRepoName = name
+		return nil
+	}
+}
+
+// FromBaseImageIndexInstance loads the provided image index for the working image index.
+// If the index is not found, it does nothing.
+func FromBaseImageIndexInstance(index v1.ImageIndex) func(options *IndexOptions) error {
+	return func(o *IndexOptions) error {
+		o.BaseIndex = index
+		return nil
+	}
+}
+
+// WithKeychain fetches Index from registry with keychain
+func WithKeychain(keychain authn.Keychain) func(options *IndexOptions) error {
+	return func(o *IndexOptions) error {
+		o.KeyChain = keychain
+		return nil
+	}
+}
+
+// WithXDGRuntimePath Saves the Index to the '`xdgPath`/manifests'
+func WithXDGRuntimePath(xdgPath string) func(options *IndexOptions) error {
+	return func(o *IndexOptions) error {
+		o.XdgPath = xdgPath
+		return nil
+	}
+}
+
+// PullInsecure If true, pulls images from insecure registry
+func PullInsecure() func(options *IndexOptions) error {
+	return func(o *IndexOptions) error {
+		o.Insecure = true
+		return nil
+	}
+}
+
+// WithFormat Create the image index with the following format
+func WithFormat(format types.MediaType) func(options *IndexOptions) error {
+	return func(o *IndexOptions) error {
+		o.Format = format
+		return nil
+	}
+}
+
+// IndexAddOptions
+
+// IndexPushOptions
+
+// If true, Deletes index from local filesystem after pushing to registry
+func WithPurge(purge bool) func(options *IndexPushOptions) error {
+	return func(a *IndexPushOptions) error {
+		a.Purge = purge
+		return nil
+	}
+}
+
+// Push the Index with given format
+func WithTags(tags ...string) func(options *IndexPushOptions) error {
+	return func(a *IndexPushOptions) error {
+		a.Tags = tags
+		return nil
+	}
+}
+
+// Push index to Insecure Registry
+func WithInsecure(insecure bool) func(options *IndexPushOptions) error {
+	return func(a *IndexPushOptions) error {
+		a.Insecure = insecure
+		return nil
+	}
+}
+
+// Push the Index with given format
+func UsingFormat(format types.MediaType) func(options *IndexPushOptions) error {
```

Comment:
Could we just set the media type when the object is instantiated, and have one media type option?